LOCALAPP.md

```diff
@@ -159,3 +122,11 @@ This would be required if your app requires a role which none of the test users
 4. Save and close the file
 5. Restart LocalTest
 
+### Tips and tricks
+
+#### Target local webpack-dev-server instead of altinncd.no
+If you're making changes to the [app frontend](https://github.com/Altinn/app-frontend-react/) locally, you're usually
+running the webpack-dev-server on port 8080 (via `yarn start`). There is no need to change the reference back and forth
+between altinncdn.no and localhost in your `App/views/Home/Index.cshtml` for every app you're running. Instead,
+copy `.env.template` to `.env` and uncomment the line where `SUB_FILTER` is set. This setting makes nginx rewrite the
+HTML for your app on the fly.
```

Comment:
What do you think about making this a checkbox backed by a cookie instead?

I have a draft version that might need some refinement (less prominent checkbox in LocalTest), and maybe make the `localhost:8080` part configurable, and check for typos.
https://github.com/Altinn/altinn-studio/compare/master...ivarne:localTest-localJs-checkbox?expand=1

Reply:
I've thought of similar things! Not exactly using a cookie, but instead of using nginx filters to redirect to localhost:8080, I day-dreamed about redirecting calls to some kind of 'Altinn DevTools' we could host on altinncdn.no instead. It could default to load the frontend from localhost (if available) and automatically fall back to the hard-coded version on altinncdn.no. We could also have it load a list of previous versions from altinncdn.no and let you swap between them (great for testing frontend bugs and/or regressions).

If we were to add to it further, we could even integrate such DevTools with Altinn Studio in some way (i.e. giving you a bit more support when working on an app locally, or having two-way communication with Studio).

Anyway, this is not what I'm tasked with.. :wink: Having to change some lines in Index.cshtml for regression testing is not the end of the world, so I guess it'll do!

Reply:
Hmm… older versions from cdn is relevant too? Maybe the cookie value should be the path for the replacement instead of a Boolean, so you get a dropdown in LocalTest for different versions (including local). That would simplify nginx config a lot (together with your trick to combine js and css).

A cdn hosted dev tool might be cool, but I don’t see anything we can’t do with nginx and LocalTest.

Reply:
Yep, a dropdown in LocalTest would work great as well! Older versions aren't always relevant, but I've worked on issues where testing in older versions is relevant in order to figure out if a bug is a regression.

Reply:
https://github.com/Altinn/altinn-studio/compare/master...ivarne:localTest-select-frontend

I have not yet implemented the logic to find urls for all old versions, but now it's easy to do from C# (not nginx), and I've moved the selection to a different page.

Because of cookies beeing url encoded, I had to use the perl mod of nginx. 

Default view in localtest
![image](https://user-images.githubusercontent.com/131616/169507640-d2c365de-efb4-4a7b-bef9-4e5026fc53f2.png)
With localhost selected
![image](https://user-images.githubusercontent.com/131616/169508383-54a29a17-cb17-4067-b97a-caf91f5fad33.png)

Version picker page (could pretty easily be extended to show older versions as well)
![image](https://user-images.githubusercontent.com/131616/169507603-a927c7a9-984b-4566-b915-21ea1b5de749.png)

Reply:
Looking good, @ivarne!